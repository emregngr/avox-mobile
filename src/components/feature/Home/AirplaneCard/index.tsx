import React, { memo, useMemo } from 'react'
import { View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'
import type { TotalAirplane } from '@/types/feature/home'

interface AirplaneCardProps {
  airplane: TotalAirplane
}

export const AirplaneCard = memo(({ airplane }: AirplaneCardProps) => {
  const { count, model } = airplane

  const modelTextProps = useMemo(
    () => ({
      children: model,
      color: 'text-90' as const,
      type: 'h4' as const,
    }),
    [model],
  )

  const countTextProps = useMemo(
    () => ({
      children: count,
      color: 'text-100' as const,
      type: 'h2' as const,
    }),
    [count],
  )

  const cardStyles = useMemo(
    () =>
      'bg-background-primary rounded-xl border border-background-quaternary shadow shadow-background-quaternary flex-row items-center justify-between mb-4 p-4',
    [],
  )

  return (
    <View className={cardStyles}>
      <ThemedText {...modelTextProps} />
      <ThemedText {...countTextProps} />
    </View>
  )
})

AirplaneCard.displayName = 'AirplaneCard'
