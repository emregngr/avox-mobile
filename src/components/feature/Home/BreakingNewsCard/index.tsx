import { Image } from 'expo-image'
import { router } from 'expo-router'
import React, { memo, useCallback, useMemo } from 'react'
import { TouchableOpacity, View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'
import type { BreakingNews } from '@/types/feature/home'
import { responsive } from '@/utils/common/responsive'

const itemWidth = responsive.deviceWidth

interface BreakingNewsCardProps {
  item: BreakingNews
}

const STATIC_STYLES = {
  cardWidth: { width: itemWidth - 32 },
  containerWidth: { width: itemWidth },
  image: {
    height: '100%' as const,
    width: '100%' as const,
  },
}

export const BreakingNewsCard = memo(({ item }: BreakingNewsCardProps) => {
  const { image, title } = item ?? {}

  const imageSource = useMemo(() => ({ uri: image }), [image])

  const handlePress = useCallback(() => {
    router.navigate({
      params: {
        item: JSON.stringify(item),
      },
      pathname: '/breaking-news-detail',
    })
  }, [item])

  return (
    <View className="items-center px-4" style={STATIC_STYLES.containerWidth}>
      <TouchableOpacity
        activeOpacity={0.7}
        className="h-64 rounded-xl overflow-hidden"
        hitSlop={20}
        onPress={handlePress}
        style={STATIC_STYLES.cardWidth}
      >
        <Image
          cachePolicy="memory-disk"
          contentFit="cover"
          source={imageSource}
          style={STATIC_STYLES.image}
          transition={0}
        />
        <View className="absolute bottom-2 mx-3 bg-background-blur px-2 py-1 overflow-hidden rounded-xl">
          <ThemedText
            color="text-100" ellipsizeMode="tail" numberOfLines={3}
            type="h4"
          >
            {title}
          </ThemedText>
        </View>
      </TouchableOpacity>
    </View>
  )
})

BreakingNewsCard.displayName = 'BreakingNewsCard'
