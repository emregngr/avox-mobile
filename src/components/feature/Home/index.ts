export * from './AirplaneCard'
export * from './BreakingNewsCard'
export * from './DestinationCard'
export * from './HomeAirlineCard'
export * from './HomeAirportCard'
export * from './HomeSlider'
export * from './NewsSection'
export * from './SectionHeader'
export * from './SectionScroll'
