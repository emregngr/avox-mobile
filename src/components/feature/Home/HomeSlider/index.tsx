import React, { memo, useCallback, useEffect, useMemo, useRef, useState } from 'react'
import type { NativeScrollEvent, NativeSyntheticEvent } from 'react-native'
import { FlatList, View } from 'react-native'

import { BreakingNewsCard } from '@/components/feature/Home/BreakingNewsCard'
import { useBatchingPeriod } from '@/hooks/batchingPeriod/useBatchingPeriod'
import type { BreakingNews } from '@/types/feature/home'
import { cn } from '@/utils/common/cn'
import { responsive } from '@/utils/common/responsive'

interface HomeSliderProps {
  breakingNews: BreakingNews[]
}
interface BreakingNewsCardProps {
  item: BreakingNews
}

interface DotProps {
  isActive: boolean
}

interface DotsContainerProps {
  breakingNews: BreakingNews[]
  currentIndex: number
}

const INITIAL_ITEMS_PER_PAGE = 4
const MAX_ITEMS_PER_BATCH = 3
const ITEM_WIDTH = responsive.deviceWidth
const WINDOW_SIZE = 7

const AUTO_SCROLL_INTERVAL = 5000

const RESTART_DELAY = 300

const Dot = memo(({ isActive }: DotProps) => {
  const dotClassName = useMemo(
    () =>
      cn('w-2 h-2 rounded-full overflow-hidden', isActive ? 'bg-onPrimary-100' : 'bg-onPrimary-50'),
    [isActive],
  )

  return (
    <View className="w-2 h-2 mx-1.5">
      <View className={dotClassName} />
    </View>
  )
})

Dot.displayName = 'Dot'

const DotsContainer = memo(({ breakingNews, currentIndex }: DotsContainerProps) => {
  const dots = useMemo(
    () => breakingNews?.map((_, i) => <Dot isActive={i === currentIndex} key={i} />),
    [breakingNews, currentIndex],
  )

  return <View className="flex-row justify-center mt-4">{dots}</View>
})

DotsContainer.displayName = 'DotsContainer'

export const HomeSlider = memo(({ breakingNews }: HomeSliderProps) => {
  const [currentIndex, setCurrentIndex] = useState<number>(0)

  const flatListRef = useRef<FlatList>(null)

  const intervalRef = useRef<NodeJS.Timeout | null>(null)

  const memoizedBreakingNews = useMemo(() => breakingNews ?? [], [breakingNews])

  const clearAutoScroll = useCallback(() => {
    if (intervalRef?.current) {
      clearInterval(intervalRef?.current)
      intervalRef.current = null
    }
  }, [])

  const startAutoScroll = useCallback(() => {
    clearAutoScroll()

    if (memoizedBreakingNews && memoizedBreakingNews?.length > 1) {
      intervalRef.current = setInterval(() => {
        setCurrentIndex(prevIndex => {
          const nextIndex = (prevIndex + 1) % memoizedBreakingNews?.length

          flatListRef?.current?.scrollToIndex({
            animated: true,
            index: nextIndex,
          })

          return nextIndex
        })
      }, AUTO_SCROLL_INTERVAL)
    }
  }, [memoizedBreakingNews, clearAutoScroll])

  const onScroll = useCallback(
    (e: NativeSyntheticEvent<NativeScrollEvent>) => {
      const scrollX = e?.nativeEvent?.contentOffset?.x
      const newIndex = Math?.round(scrollX / ITEM_WIDTH)

      if (newIndex !== currentIndex && newIndex >= 0 && newIndex < memoizedBreakingNews?.length) {
        setCurrentIndex(newIndex)
      }
    },
    [currentIndex, memoizedBreakingNews?.length],
  )

  const onMomentumScrollEnd = useCallback(
    (e: NativeSyntheticEvent<NativeScrollEvent>) => {
      const scrollX = e?.nativeEvent?.contentOffset?.x
      const newIndex = Math?.round(scrollX / ITEM_WIDTH)

      if (newIndex !== currentIndex && newIndex >= 0 && newIndex < memoizedBreakingNews?.length) {
        setCurrentIndex(newIndex)
      }

      setTimeout(() => {
        startAutoScroll()
      }, RESTART_DELAY)
    },
    [currentIndex, memoizedBreakingNews?.length, startAutoScroll],
  )

  const onScrollBeginDrag = useCallback(() => {
    clearAutoScroll()
  }, [clearAutoScroll])

  const BATCHING_PERIOD = useBatchingPeriod()

  useEffect(() => {
    startAutoScroll()

    return () => clearAutoScroll()
  }, [startAutoScroll, clearAutoScroll])

  const shouldRender = useMemo(
    () => memoizedBreakingNews && memoizedBreakingNews?.length > 0,
    [memoizedBreakingNews],
  )

  const renderItem = useCallback(
    ({ item }: BreakingNewsCardProps) => <BreakingNewsCard item={item} />,
    [],
  )

  const keyExtractor = useCallback((item: BreakingNews) => item?.id?.toString(), [])

  const getItemLayout = useCallback(
    (_: ArrayLike<BreakingNews> | null | undefined, index: number) => ({
      index,
      length: ITEM_WIDTH,
      offset: ITEM_WIDTH * index,
    }),
    [],
  )

  if (!shouldRender) {
    return null
  }

  return (
    <View>
      <FlatList
        bounces={false}
        data={memoizedBreakingNews}
        decelerationRate="fast"
        getItemLayout={getItemLayout}
        initialNumToRender={INITIAL_ITEMS_PER_PAGE}
        keyExtractor={keyExtractor}
        maxToRenderPerBatch={MAX_ITEMS_PER_BATCH}
        onMomentumScrollEnd={onMomentumScrollEnd}
        onScroll={onScroll}
        onScrollBeginDrag={onScrollBeginDrag}
        overScrollMode="never"
        ref={flatListRef}
        renderItem={renderItem}
        scrollEventThrottle={16}
        showsHorizontalScrollIndicator={false}
        updateCellsBatchingPeriod={BATCHING_PERIOD}
        windowSize={WINDOW_SIZE}
        disableIntervalMomentum
        horizontal
        pagingEnabled
      />
      <DotsContainer breakingNews={memoizedBreakingNews} currentIndex={currentIndex} />
    </View>
  )
})

HomeSlider.displayName = 'HomeSlider'
