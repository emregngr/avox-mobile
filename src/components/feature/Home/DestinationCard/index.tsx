import React, { memo, useMemo } from 'react'
import { View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import type { PopularDestination } from '@/types/feature/home'
import { formatNumber } from '@/utils/feature/formatNumber'

interface DestinationCardProps {
  destination: PopularDestination
}

export const DestinationCard = memo(({ destination }: DestinationCardProps) => {
  const { selectedLocale } = useLocaleStore()

  const { country, distance_km, flight_count, route, type } = destination

  const formattedFlightCount = useMemo(() => formatNumber(flight_count), [flight_count])

  const formattedDistance = useMemo(() => formatNumber(distance_km), [distance_km])

  const localeStrings = useMemo(
    () => ({
      distance: getLocale('distance'),
      flightPerYear: getLocale('flight/year'),
      km: getLocale('km'),
    }),
    [selectedLocale],
  )

  const distanceWithUnit = useMemo(
    () => `${formattedDistance} ${localeStrings.km}`,
    [formattedDistance, localeStrings.km],
  )

  return (
    <View className="w-36 mb-4 bg-background-secondary rounded-xl border border-background-quaternary shadow shadow-background-quaternary">
      <View className="bg-background-primary rounded-xl overflow-hidden relative px-2 border-b border-background-quaternary w-full h-44 justify-center">
        <ThemedText
          className="mx-2" color="tertiary-100" type="h4"
          center
        >
          {route}
        </ThemedText>

        <View className="bg-primary-100 px-2 py-1 rounded-xl overflow-hidden absolute bottom-2 left-2">
          <ThemedText
            color="text-100" ellipsizeMode="tail" numberOfLines={1}
            type="button2"
          >
            {type}
          </ThemedText>
        </View>
      </View>

      <View className="px-3 py-3">
        <View className="h-32 justify-between">
          <ThemedText
            className="mb-2"
            color="text-100"
            ellipsizeMode="tail"
            numberOfLines={2}
            type="body3"
          >
            {country}
          </ThemedText>

          <View className="absolute bottom-10">
            <ThemedText
              color="text-70" ellipsizeMode="tail" numberOfLines={1}
              type="body4"
            >
              {localeStrings.flightPerYear}
            </ThemedText>
            <ThemedText
              color="text-90" ellipsizeMode="tail" numberOfLines={1}
              type="body3"
            >
              {formattedFlightCount}
            </ThemedText>
          </View>

          <View className="absolute bottom-0">
            <ThemedText
              color="text-70" ellipsizeMode="tail" numberOfLines={1}
              type="body4"
            >
              {localeStrings.distance}
            </ThemedText>
            <ThemedText
              color="text-90" ellipsizeMode="tail" numberOfLines={1}
              type="body3"
            >
              {distanceWithUnit}
            </ThemedText>
          </View>
        </View>
      </View>
    </View>
  )
})

DestinationCard.displayName = 'DestinationCard'
