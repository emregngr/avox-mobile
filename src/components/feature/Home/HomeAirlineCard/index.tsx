import { Image } from 'expo-image'
import { router } from 'expo-router'
import React, { memo, useCallback, useMemo } from 'react'
import { Platform, TouchableOpacity, View } from 'react-native'
import { TestIds } from 'react-native-google-mobile-ads'

import { ThemedText } from '@/components/common/ThemedText'
import { FavoriteButton } from '@/components/feature/FavoriteButton'
import { useInterstitialAdHandler } from '@/hooks/advertisement/useInterstitialAdHandler'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import type { Airline } from '@/types/feature/airline'
import { AnalyticsService } from '@/utils/common/analyticsService'
import type { AirlineType } from '@/utils/feature/getBadge'
import { getAirlineBadge } from '@/utils/feature/getBadge'

interface HomeAirlineCardProps {
  airline: Airline
}

const AD_UNIT_ID = __DEV__
  ? TestIds.INTERSTITIAL
  : Platform.OS === 'ios'
    ? 'ca-app-pub-4123130377375974/1756124081'
    : 'ca-app-pub-4123130377375974/2330839152'

const STATIC_STYLES = {
  badge: {
    bottom: 4,
    height: 40,
    left: 4,
    position: 'absolute' as const,
    width: 40,
  },
  title: {
    height: 130,
  },
}

export const HomeAirlineCard = memo(({ airline }: HomeAirlineCardProps) => {
  const { selectedLocale } = useLocaleStore()

  const {
    iataCode,
    icaoCode,
    id,
    name,
    operations: {
      businessModel,
      businessType,
      country,
      hub: { city },
      region,
    },
  } = airline ?? {}

  const badge = useMemo(() => getAirlineBadge(businessType as AirlineType), [businessType])

  const locationText = useMemo(() => `${city}, ${country}, ${region}`, [city, country, region])

  const classNames = useMemo(
    () => ({
      cargoContainer: 'bg-error px-2 py-1 rounded-xl overflow-hidden absolute top-2 left-2',
      container:
        'w-36 mb-4 bg-background-secondary rounded-xl border border-background-quaternary shadow shadow-background-quaternary',
      content: 'px-3 py-3',
      contentInner: 'h-28 justify-between',
      header:
        'bg-background-primary rounded-t-xl overflow-hidden w-full justify-center overflow-hidden',
      icaoCodeContainer:
        'bg-primary-100 px-2 py-1 rounded-xl overflow-hidden absolute bottom-2 right-2',
      locationText: 'mt-1',
    }),
    [],
  )

  const cargoLabel = useMemo(
    () =>
      businessModel === 'cargo' ? (
        <View className={classNames.cargoContainer}>
          <ThemedText color="text-100" type="body3">
            {getLocale('cargo')}
          </ThemedText>
        </View>
      ) : null,
    [businessModel, classNames.cargoContainer],
  )

  const logAirlineCardPress = useCallback(async () => {
    await AnalyticsService.sendEvent('airline_card_press', {
      airline_id: id,
      airline_name: name,
      iata_code: iataCode,
      user_locale: selectedLocale,
    })
  }, [id, name, iataCode, selectedLocale])

  const handlePress = useCallback(() => {
    router.navigate({
      params: {
        airline: JSON.stringify(airline),
      },
      pathname: '/airline-detail',
    })
  }, [airline])

  const { showInterstitialAd } = useInterstitialAdHandler({
    adUnitId: AD_UNIT_ID,
  })

  const onCardPress = useCallback(() => {
    logAirlineCardPress()
    handlePress()
    showInterstitialAd()
  }, [logAirlineCardPress, handlePress, showInterstitialAd])

  return (
    <TouchableOpacity
      activeOpacity={0.7}
      className={classNames.container}
      hitSlop={20}
      onPress={onCardPress}
    >
      <View className={classNames.header} style={STATIC_STYLES.title}>
        <ThemedText color="tertiary-100" type="title" center>
          {iataCode}
        </ThemedText>

        <Image
          cachePolicy="memory-disk"
          contentFit="contain"
          source={badge}
          style={STATIC_STYLES.badge}
          transition={0}
        />

        <View className={classNames.icaoCodeContainer}>
          <ThemedText color="text-100" type="button2">
            {icaoCode}
          </ThemedText>
        </View>

        {cargoLabel}

        <FavoriteButton id={String(id)} type="airline" />
      </View>

      <View className={classNames.content}>
        <View className={classNames.contentInner}>
          <ThemedText
            color="text-100" ellipsizeMode="tail" numberOfLines={2}
            type="body3"
          >
            {name}
          </ThemedText>

          <ThemedText
            className={classNames.locationText}
            color="text-90"
            ellipsizeMode="tail"
            numberOfLines={2}
            type="body3"
          >
            {locationText}
          </ThemedText>
        </View>
      </View>
    </TouchableOpacity>
  )
})

HomeAirlineCard.displayName = 'HomeAirlineCard'
