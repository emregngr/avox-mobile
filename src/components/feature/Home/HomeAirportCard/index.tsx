import { Image } from 'expo-image'
import { router } from 'expo-router'
import React, { memo, useCallback, useMemo } from 'react'
import { TouchableOpacity, View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'
import { FavoriteButton } from '@/components/feature/FavoriteButton'
import type { Airport } from '@/types/feature/airport'
import getAirportImage from '@/utils/feature/getAirportImage'
import type { AirportType } from '@/utils/feature/getBadge'
import { getAirportBadge } from '@/utils/feature/getBadge'

interface HomeAirportCardProps {
  airport: Airport
}

export const HomeAirportCard = memo(({ airport }: HomeAirportCardProps) => {
  const {
    iataCode,
    icaoCode,
    id,
    name,
    operations: {
      airportType,
      country,
      location: { city },
      region,
    },
  } = airport || {}

  const images = useMemo(() => {
    const airportImage = getAirportImage(airportType as AirportType)
    const airportBadge = getAirportBadge(airportType as AirportType)

    return {
      airportBadge,
      airportImage,
    }
  }, [airportType])

  const handlePress = useCallback(() => {
    router.navigate({
      params: {
        airport: JSON.stringify(airport),
      },
      pathname: '/airport-detail',
    })
  }, [airport])

  const locationText = useMemo(() => `${city}, ${country}, ${region}`, [city, country, region])

  const styles = useMemo(
    () => ({
      badgeImage: {
        height: 40,
        left: 4,
        position: 'absolute' as const,
        top: 4,
        width: 40,
      },
      mainImage: {
        height: 130,
        width: '100%' as const,
      },
    }),
    [],
  )

  const classNames = useMemo(
    () => ({
      container:
        'w-36 mb-4 bg-background-secondary rounded-xl border border-background-quaternary shadow shadow-background-quaternary',
      content: 'px-3 py-3',
      contentInner: 'h-28 justify-between',
      iataCodeContainer:
        'bg-primary-100 px-2 py-1 rounded-xl overflow-hidden absolute bottom-2 left-2',
      icaoCodeContainer:
        'bg-primary-100 px-2 py-1 rounded-xl overflow-hidden absolute bottom-2 right-2',
      imageContainer: 'rounded-t-xl w-full justify-center overflow-hidden',
      locationText: 'mt-1',
    }),
    [],
  )

  return (
    <TouchableOpacity
      activeOpacity={0.7}
      className={classNames.container}
      hitSlop={20}
      onPress={handlePress}
    >
      <View className={classNames.imageContainer}>
        <Image
          cachePolicy="memory-disk"
          contentFit="cover"
          source={images.airportImage}
          style={styles.mainImage}
          transition={0}
        />

        <Image
          cachePolicy="memory-disk"
          contentFit="contain"
          source={images.airportBadge}
          style={styles.badgeImage}
          transition={0}
        />

        <View className={classNames.iataCodeContainer}>
          <ThemedText color="text-100" type="button2">
            {iataCode}
          </ThemedText>
        </View>

        <View className={classNames.icaoCodeContainer}>
          <ThemedText color="text-100" type="button2">
            {icaoCode}
          </ThemedText>
        </View>

        <FavoriteButton id={String(id)} type="airport" />
      </View>

      <View className={classNames.content}>
        <View className={classNames.contentInner}>
          <ThemedText
            color="text-100" ellipsizeMode="tail" numberOfLines={2}
            type="body3"
          >
            {name}
          </ThemedText>

          <ThemedText
            className={classNames.locationText}
            color="text-90"
            ellipsizeMode="tail"
            numberOfLines={2}
            type="body3"
          >
            {locationText}
          </ThemedText>
        </View>
      </View>
    </TouchableOpacity>
  )
})

HomeAirportCard.displayName = 'HomeAirportCard'
