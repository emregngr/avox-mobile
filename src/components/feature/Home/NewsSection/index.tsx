import { View } from 'react-native'

import { HomeSlider } from '@/components/feature/Home/HomeSlider'
import { SectionHeader } from '@/components/feature/Home/SectionHeader'
import { getLocale } from '@/locales/i18next'
import type { BreakingNewsType } from '@/types/feature/home'

interface NewsSectionProps {
  breakingNews: BreakingNewsType[]
}

export const NewsSection = ({ breakingNews }: NewsSectionProps) => (
  <View className="mt-3 mb-8">
    <SectionHeader title={getLocale('news')} />
    <HomeSlider breakingNews={breakingNews} />
  </View>
)
