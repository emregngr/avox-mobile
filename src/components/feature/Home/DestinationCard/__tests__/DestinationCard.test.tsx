import { render } from '@testing-library/react-native'
import React from 'react'

import { DestinationCard } from '@/components/feature/Home/DestinationCard'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import type { PopularDestinationType } from '@/types/feature/home'

jest.mock('@/locales/i18next')

const mockedGetLocale = getLocale as jest.MockedFunction<typeof getLocale>

jest.mock('@/store/locale')

const mockedUseLocaleStore = useLocaleStore as jest.MockedFunction<typeof useLocaleStore>

jest.mock('@/components/common/ThemedText', () => {
  const { Text } = require('react-native')

  return {
    ThemedText: ({ children, ...props }: { children: string }) => (
      <Text {...props}>{children}</Text>
    ),
  }
})

const mockedDestination: PopularDestinationType = {
  id: '1',
  country: 'Türkiye',
  destinations_type: 'International',
  distance_km: 12345,
  flight_count: 678,
  route: 'IST → JFK',
}

beforeEach(() => {
  mockedUseLocaleStore.mockReturnValue({ selectedLocale: 'en' })

  mockedGetLocale.mockImplementation((key: string) => {
    const localeMap: Record<string, string> = {
      distance: 'Distance',
      'flight/year': 'Flights per year',
      km: 'km',
      domestic: 'Domestic',
      international: 'International',
    }
    return localeMap[key] || key
  })
})

describe('DestinationCard', () => {
  it('renders route and country correctly', () => {
    const { getByText } = render(<DestinationCard destination={mockedDestination} />)

    expect(getByText('IST → JFK')).toBeTruthy()
    expect(getByText('Türkiye')).toBeTruthy()
  })

  it('renders formatted flight count and distance with unit', () => {
    const { getByText } = render(<DestinationCard destination={mockedDestination} />)

    expect(getByText('678')).toBeTruthy()
    expect(getByText('12.345 km')).toBeTruthy()
  })

  it('renders localized labels correctly', () => {
    const { getByText } = render(<DestinationCard destination={mockedDestination} />)

    expect(getByText('Flights per year')).toBeTruthy()
    expect(getByText('Distance')).toBeTruthy()
    expect(getByText('International')).toBeTruthy()
  })

  it('should render the DestinationCard successfully', () => {
    render(<DestinationCard destination={mockedDestination} />)

    const { toJSON } = render(<DestinationCard destination={mockedDestination} />)

    expect(toJSON()).toMatchSnapshot()
  })
})

describe('DestinationCard Component Snapshot', () => {
  it('should render the DestinationCard Component successfully', () => {
    const { toJSON } = render(<DestinationCard destination={mockedDestination} />)

    expect(toJSON()).toMatchSnapshot()
  })
})
