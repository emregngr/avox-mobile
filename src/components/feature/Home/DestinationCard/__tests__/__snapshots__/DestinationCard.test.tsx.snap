// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`DestinationCard Component Snapshot should render the DestinationCard Component successfully 1`] = `
<View
  className="w-36 mb-4 bg-background-secondary rounded-xl border border-background-quaternary shadow shadow-background-quaternary"
>
  <View
    className="bg-background-primary rounded-xl overflow-hidden relative px-2 border-b border-background-quaternary w-full h-44 justify-center"
  >
    <Text
      center={true}
      className="mx-2"
      color="tertiary-100"
      type="h4"
    >
      IST → JFK
    </Text>
    <View
      className="bg-primary-100 px-2 py-1 rounded-xl overflow-hidden absolute bottom-2 left-2"
    >
      <Text
        color="text-100"
        ellipsizeMode="tail"
        numberOfLines={1}
        type="button2"
      >
        International
      </Text>
    </View>
  </View>
  <View
    className="px-3 py-3"
  >
    <View
      className="h-32 justify-between"
    >
      <Text
        className="mb-2"
        color="text-100"
        ellipsizeMode="tail"
        numberOfLines={2}
        type="body3"
      >
        Türkiye
      </Text>
      <View
        className="absolute bottom-10"
      >
        <Text
          color="text-70"
          ellipsizeMode="tail"
          numberOfLines={1}
          type="body4"
        >
          Flights per year
        </Text>
        <Text
          color="text-90"
          ellipsizeMode="tail"
          numberOfLines={1}
          type="body3"
        >
          678
        </Text>
      </View>
      <View
        className="absolute bottom-0"
      >
        <Text
          color="text-70"
          ellipsizeMode="tail"
          numberOfLines={1}
          type="body4"
        >
          Distance
        </Text>
        <Text
          color="text-90"
          ellipsizeMode="tail"
          numberOfLines={1}
          type="body3"
        >
          12.345 km
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`DestinationCard should render the DestinationCard successfully 1`] = `
<View
  className="w-36 mb-4 bg-background-secondary rounded-xl border border-background-quaternary shadow shadow-background-quaternary"
>
  <View
    className="bg-background-primary rounded-xl overflow-hidden relative px-2 border-b border-background-quaternary w-full h-44 justify-center"
  >
    <Text
      center={true}
      className="mx-2"
      color="tertiary-100"
      type="h4"
    >
      IST → JFK
    </Text>
    <View
      className="bg-primary-100 px-2 py-1 rounded-xl overflow-hidden absolute bottom-2 left-2"
    >
      <Text
        color="text-100"
        ellipsizeMode="tail"
        numberOfLines={1}
        type="button2"
      >
        International
      </Text>
    </View>
  </View>
  <View
    className="px-3 py-3"
  >
    <View
      className="h-32 justify-between"
    >
      <Text
        className="mb-2"
        color="text-100"
        ellipsizeMode="tail"
        numberOfLines={2}
        type="body3"
      >
        Türkiye
      </Text>
      <View
        className="absolute bottom-10"
      >
        <Text
          color="text-70"
          ellipsizeMode="tail"
          numberOfLines={1}
          type="body4"
        >
          Flights per year
        </Text>
        <Text
          color="text-90"
          ellipsizeMode="tail"
          numberOfLines={1}
          type="body3"
        >
          678
        </Text>
      </View>
      <View
        className="absolute bottom-0"
      >
        <Text
          color="text-70"
          ellipsizeMode="tail"
          numberOfLines={1}
          type="body4"
        >
          Distance
        </Text>
        <Text
          color="text-90"
          ellipsizeMode="tail"
          numberOfLines={1}
          type="body3"
        >
          12.345 km
        </Text>
      </View>
    </View>
  </View>
</View>
`;
